### DISCUSSION TITLE
Lazer caps FPS at 60 when monitor refresh rate is 360Hz

### DISCUSSION THREAD

**Discussion by AwesomeRedDragon** [ppy/osu#29181] [Q&A] [answered]
### Type

Performance

### Bug description

I have a laptop that lets me switch if I want refresh rate to be 60Hz or 360Hz. On 60Hz, the game works normally, where 2x refresh rate gives me 120fps, so on and so forth, but on 360Hz, no matter what setting I pick FPS is capped at 60.

### Screenshots or videos

https://github.com/user-attachments/assets/3ac49970-1758-4f7d-a79d-353732c5b434
Started video on 60Hz, switched back and forth. Only number that changes is "draw".

### Version

2024.726.0-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16395195/compressed-logs.zip)
Opened lazer on 60Hz, sat on home screen for a bit, then switched to 360Hz to show frame drop before exporting. Not sure if anything shows besides the frame rate dropping

**Comment by smoogipoo** [comment]
Can you test out setting the environment variable `OSU_SDL3=1`?

**Reply by AwesomeRedDragon** [reply]
Where can I find and change environment variables?

**Reply by smoogipoo** [reply]
Two ways.

1) Add it globally. Press windows key, search "environment variables", click the "environment variables" button, add a new user environment variable.

2) Locally, via Powershell. Navigate to the osu! directory (right click -> open file location). Then open a Powershell there and type:
```pwsh
$env:OSU_SDL3="1"

.\osu!.exe
```

If (1) from above, then I recommend resetting it if it doesn't resolve the issue.

**Reply by AwesomeRedDragon** [reply]
That uncapped my fps while at 60hz refresh rate, but still kept me at 60fps for 360hz refresh rate. Logs (switching back and forth a few times):
[compressed-logs.zip](https://github.com/user-attachments/files/16422123/compressed-logs.zip)

**Reply by smoogipoo** [reply]
It doesn't look like the environment variable took effect, unfortunately. I'll look to add a setting for this in the next release.

**Reply by AwesomeRedDragon** [reply]
I just went back to the environment variables panel and didn't see the variable added, for whatever reason, so I changed it again and redid the logs, sorry about that. Still didn't fix the issue, but maybe gives you more insight?
[compressed-logs.zip](https://github.com/user-attachments/files/16422168/compressed-logs.zip)

**Reply by smoogipoo** [reply]
That looks better. Very weird, I wonder if it's something with Razer's software, or some sort of adaptive refresh rate.

I noticed that in one of your logs, it looked like you started the game while already in 360Hz mode, does it work correctly in that case?

What happens if you use the windows setting (right click on desktop -> display settings (and then you probably have to go into advanced settings depending on your OS...)?

**Reply by AwesomeRedDragon** [reply]
I just tried launching lazer twice, once on each setting from the windows advanced display settings (one shows as 60.05Hz, the other as 360Hz), and launching on 60.05 had my fps at 240 while launching on 360 had me at 60 (my setting was 4x refresh rate both times, so both should have been 240)

**Reply by AwesomeRedDragon** [reply, marked as answer]
Just wanted to resolve this in case anyone comes across the same issue - I just installed intel display driver update 31.0.101.4255 and this issue no longer occurs.
